src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Selectors/SecurityTokenAuthenticator.cs

```diff
@@ -42,6 +50,14 @@ public ReadOnlyCollection<IAuthorizationPolicy> ValidateToken(SecurityToken toke
         }
 
         protected abstract bool CanValidateTokenCore(SecurityToken token);
-        protected abstract ReadOnlyCollection<IAuthorizationPolicy> ValidateTokenCore(SecurityToken token);
+
+        [Obsolete("Implementers should override ValidateTokenCoreAsync.")]
+        protected virtual ReadOnlyCollection<IAuthorizationPolicy> ValidateTokenCore(SecurityToken token) => throw new NotImplementedException(SR.SynchronousTokenValidationIsDeprecated);
```

Comment:
@mconnew do you still have any concern with this `abstract` made `virtual` ?

Reply:
I checked with some experts on the matter and abstract -> virtual is safe from a backwards compatibility standpoint.